lineapy/api/models/linea_artifact.py

```diff
@@ -105,36 +103,118 @@ def get_value(self) -> object:
         """
         Get and return the value of the artifact
         """
-        pickle_filename = self.db.get_node_value_path(
-            self._node_id, self._execution_id
-        )
-        if pickle_filename is None:
+        metadata = self.get_metadata()
+        linea_metadata = metadata["lineapy"]
+        saved_filepath = linea_metadata.storage_path
+        if saved_filepath is None:
             return None
         else:
-            # TODO - set unicode etc here
             track(GetValueEvent(has_value=True))
+            # read from mlflow
+            if "mlflow" in metadata.keys():
+                return read_mlflow(metadata["mlflow"])
+
+            # read from lineapy
+            return self._read_pickle(saved_filepath)
+
+    @lru_cache(maxsize=None)
+    def _get_storage_path(self) -> Optional[str]:
+        return self.db.get_node_value_path(self._node_id, self._execution_id)
+
+    @lru_cache(maxsize=None)
+    def get_metadata(self, lineapy_only: bool = False) -> ArtifactInfo:
```

Comment:
is this storage metadata only? If so  lets rename to get_storage_metadata.

Reply:
No, just for MLflow all the metadata are storage related.